src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,214 @@
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectedFilingDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
+    return startDateMax
+  }
+
+  /** The maximum start date string. */
+  protected get startDateMaxStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMax)
+  }
+
+  /** The recognition date or business start date-time string. */
+  protected get recognitionDateTime (): string {
+    if (this.isBenIaCorrectionFiling) {
+      if (this.getCorrectedFilingDate) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectedFilingDate, true)
+      }
+    }
+    if (
+      this.isFirmCorrectionFiling ||
+      this.isAlterationFiling ||
+      this.isChangeRegFiling ||
+      this.isFirmConversionFiling
+    ) {
+      if (this.getCorrectedFilingDate) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectedFilingDate, true)
+      } else if (this.getBusinessFoundingDate) {
+        return this.apiToPacificDateLong(this.getBusinessFoundingDate)
+      }
+    }
+    return 'Unknown'
+  }
+
+  protected onChangeClicked (): void {
+    this.onEditMode = true
+  }
+
+  /** Called to add a new date. */
+  protected onOkClicked (date: string): void {
+    if (date) {
+      this.newCorrectedStartDate = date
+    }
+  }
+
+  protected onDoneClicked (): void {
+    if (this.newCorrectedStartDate) {
+      if (this.yyyyMmDdToPacificDate(this.newCorrectedStartDate, true) !==
+        this.apiToPacificDateLong(this.getBusinessFoundingDate)) {
+        this.setCorrectionFilingDate(this.newCorrectedStartDate)
+        this.isCorrected = true
+      } else {
+        this.newCorrectedStartDate = undefined
+      }
+    }
+    this.onEditMode = false
+  }
+
+  protected onCancelClicked (): void {
+    this.onEditMode = false
+  }
+
+  protected onUndoClicked (): void {
+    this.setCorrectionFilingDate(undefined)
+    this.newCorrectedStartDate = undefined
+    this.isCorrected = false
+  }
+
+  @Watch('onEditMode', { immediate: true })
+  protected syncValidity (): void {
+    const isValid = !this.onEditMode
```

Comment:
The only time this section is invalid is when you are picking a date, otherwise you cannot select invalid dates.

Reply:
What if you open the date picker and then click close/cancel?

Reply:
Then there won't be a corrected filing date and the correction will keep the business founding date.

Reply:
So it's never blank?

Reply:
The date displayed should never be blank, however if there is no business founding date or corrected filing date then it will default to Unkown.

Reply:
Should the component should be in error if there is no date?

Reply:
I am not sure I will ask would that be Mihai or Linda

Reply:
Mihai is our BA and can answer business questions.

Dave, as our current PO, can answer business questions.

Yui might be a good choice to ask, as this is a UI interaction, and he can ask BA/PO if it depends on a business requirement.